### PR TITLE
Error utility

### PR DESCRIPTION
**Pull request by bolt12**
Tackling #296 !

### CODE REVIEW

File: src/Polysemy/Error.hs [hunk 1]
```diff
@@ -148,6 +151,45 @@ note e Nothing  = throw e
 note _ (Just a) = pure a
 {-# INLINABLE note #-}
 
+------------------------------------------------------------------------------
+-- | Similar to @'catch'@, but returns an @'Either'@ result which is (@'Right' a@) 
+-- if no exception of type @e@ was raised, or (@'Left' ex@) if an exception of type 
+-- @e@ was raised and its value is @ex@. If any other type of exception is raised 
+-- than it will be propogated up to the next enclosing exception handler.
```
[the remark this answers is not in the document]
Comment (bolt12) [reply src/Polysemy/Error.hs]: Please tell me if it's alright now :D

[the remark this answers is not in the document]
Comment (KingoftheHomeless) [reply src/Polysemy/Error.hs]: You forgot the second "raised" in `try`.

**Comment by bolt12** [comment]
Hey @KingoftheHomeless, sorry for the delay! I hope everything is flawless now.

**Comment by KingoftheHomeless** [comment]
Thanks!

**Pull request merged**